homeassistant/components/nuki/lock.py

```diff
@@ -189,3 +244,20 @@ def open(self, **kwargs):
 
     def lock_n_go(self, unlatch):
         """Stub service."""
+
+    async def handle_callback(self, event):
+        """Handle webhook events."""
+        await super().handle_callback(event)
+
+        if (
+            self._nuki_device.ring_action_state is True
+            and self._last_ring_action_timestamp
+            != self._nuki_device.ring_action_timestamp
+        ):
+            event_data = {
+                ATTR_NUKI_ID: self._nuki_device.nuki_id,
+                ATTR_RING_ACTION_TIMESTAMP: self._nuki_device.ring_action_timestamp,
+            }
+            self.hass.bus.fire(EVENT_NUKI_BELL_RING, event_data)
```

Comment:
We don't fire custom events from inside entities. They already indirectly fire state changed events.

Reply:
Ok, so I should add the attributes `ring_action_state` and `ring_action_timestamp` to the `NukiOpenerEntity` `device_state_attributes` instead and remove the event completely?

Reply:
What does this data represent?

Reply:
The Nuki Opener can be connected to an intercom system. It detects when someone rings the bell.

From the Nuki Bridge HTTP API Doc:
`ringactionState`: Flag indicating if a ring-action is currently occuring or not (reset after 30 seconds)
`ringactionTimestamp`: Timestamp of the last ring-action

Unfortunately the `ringactionState` didn't get reset via the callback, at least in my tests.

Anyway, listening on state changes of the `ringactionTimestamp` attribute should basically be equivalent to my event implementations.

Reply:
It sounds like the ring signal is a stateless occurrence, so then it should be represented by a custom event and not a state.

So then we should fire a custom event but not from the entity or lock platform, but from another integration module, eg the init module.

So this will require some refactor. I suggest to do that separately from the webhook addition.

Reply:
Makes sense. Thanks for your feedback.
I will remove the event from this PR for now.